app/internal/import_file.py

```diff
@@ -0,0 +1,178 @@
+            date_n_count[event["Date"]] += 1
+            if date_n_count[event["Date"]] > same_date_counter:
+                same_date_counter = date_n_count[event["Date"]]
+        else:
+            date_n_count[event["Date"]] = 1
+    return same_date_counter <= max_event_start_date
+
+
+def import_txt_file(txt_file: str) -> List[Dict[str, Union[str, Any]]]:
+    calendar_content = []
+    with open(txt_file, "r") as text:
+        events = text.readlines()
+    for event in events:
+        if not check_validity_of_txt(event):
+            return list()
+        head, content, event_date = event.split(", ")
+        if not check_date_in_range(event_date.replace("\n", "")):
+            return list()
+        event_date = datetime.datetime.strptime(event_date.replace("\n", ""),
+                                                "%m-%d-%Y")
+        calendar_content.append({"Head": head,
+                                 "Content": content,
+                                 "Date": event_date})
+    return calendar_content
+
+
+def import_ics_file(ics_file: str) -> List[Dict[str, Union[str, Any]]]:
+    calendar_content = []
+    with open(ics_file, "r") as ics:
+        try:
+            calendar_read = Calendar.from_ical(ics.read())
+        except (IndexError, ValueError):
+            return list()
+        for component in calendar_read.walk():
+            if component.name == "VEVENT":
+                if str(component.get('summary')) is None or \
+                   component.get('dtstart') is None or \
+                   not check_date_in_range(component.get('dtstart').dt):
+                    return list()
+                else:
+                    calendar_content.append({
+                        "Head": str(component.get('summary')),
+                        "Content": str(component.get('description')),
+                        "Date": component.get('dtstart').dt
+                        .replace(tzinfo=None)
+                    })
+    return calendar_content
+
+
+def move_events_to_db(events: List[Dict[str, Union[str, Any]]],
+                      user_id: int,
+                      session: SessionLocal = SessionLocal()) -> None:
+    """insert the events into Event table"""
+    for event in events:
+        event = Event(
+            title=event["Head"],
+            content=event["Content"],
+            date=event["Date"],
+            owner_id=user_id
+        )
+        session.add(event)
+    session.commit()
+    session.close()
+
+
+def user_click_import(file: str, user_id: int,
+                      session: SessionLocal = SessionLocal()) -> str:
+    """
+    when user choose a file and click import, we are checking the file
+    and if everything is ok we will insert the data to DB
+    """
+    if before_import_checking(file):
+        if file.lower().endswith(VALID_FILE_EXTENSION[-1]):
+            import_file = import_ics_file(file)
+        else:
+            import_file = import_txt_file(file)
```

Comment:
This is the function for txt and csv files?

Reply:
yes, import_txt_file(file) handles txt and csv files